Kernel/Arch/i686/Memory.i686.hpp

```diff
@@ -167,21 +183,38 @@ class Address
         uintptr_t val;
     } m_addr;
 };
+#endif
 
 /**
  * @brief Invalidate the page at the given address. Implementations are architecture
  * specific.
  *
  * @param addr Address of page to be invalidated
  */
-inline void pageInvalidate(void* addr)
+static inline void pageInvalidate(void* addr)
 {
-   asm volatile("invlpg (%0)" ::"r" (addr) : "memory");
+   asm volatile(
+       "invlpg (%0)"
+       :
+       : "r" (addr)
+       : "memory"
+    );
 }
 
-inline void setPageDirectory(size_t page_dir)
+/**
+ * @brief Writes the address of the page directory to CR3.
+ * Does not enable paging.
+ *
+ * @param pageDirPtr Address of page directory structure to be used
+ */
+__attribute__((always_inline))
+static inline void setPageDirectory(uintptr_t pageDirPtr)
 {
-    asm volatile("mov %0, %%cr3" ::"b"(page_dir));
+    asm volatile(
+        "mov %0, %%cr3"
+        :
+        : "b" (pageDirPtr)
```

Comment:
I know it wasn't here before, but I feel like this should have the `memory` clobber since it uh kinda changes the memory layout.

Reply:
Actually this is probably something that should be resolved separately